.github/workflows/_tf_plan_apply.yml

```diff
@@ -84,3 +106,32 @@ jobs:
       - name: run plan/apply on GCP
         if: inputs.cloud == 'gcp'
         run: echo GCP placeholder
+
+  cleanup:
+    needs: [terraform]
+    name: cleanup infrastructure in case of CI failure
+    if: cancelled() || failure()
+    permissions:
+      contents: read
+      id-token: write
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: checkout code
+        uses: actions/checkout@v3
+
+      - name: cleanup Azure Subscription
+        if: inputs.cloud == 'azure'
+        uses: ./.github/actions/sub_cleanup
```

Comment:
I think this workflows needs to be added to the Azure TF repo - it currently isn't.

Reply:
yeah, but it's a chicken-egg problem, this piece of code needs an update to the public repo, but the public repo needs this release to use it :D I've figured out that since we use versioning, I'll push these 1st as a major and the prepare the Azure changes

Reply:
clear :D !